lib/ipam/ipam.go

```diff
@@ -1414,24 +1414,62 @@ func (c ipamClient) SetIPAMConfig(ctx context.Context, cfg IPAMConfig) error {
 		return errors.New("Cannot disable 'StrictAffinity' and 'AutoAllocateBlocks' at the same time")
 	}
 
-	allObjs, err := c.client.List(ctx, model.BlockListOptions{}, "")
-	if err != nil {
-		return err
+	// When moving from StrictAffinity true to false, there is a need
```

Comment:
So, I spoke with @robbrockbank about this and I think we can simplify this a bit.

The reason we added the StrictAffinity field on the block object originally was to prevent extra datastore requests to retrieve the IPAMConfig object, except when allocating new blocks.

However, nowadays, the code is already querying the IPAMConfig object every time we need it, so the information in the block isn't serving that purpose. 

It _maybe_ has some minimal value in that writes to block are atomic, and so you won't get two clients seeing a different value for `StrictAffinity` for a given block (e.g., client1 queries IPAMConfig and it shows "true", client2 queries it and it shows "false"), but I don't think we need to worry about that. if the IPAMConfig is _changing_, then of course the user should expect that maybe some allocations got borrowed. 

So, I think we can do the following:
- Remove this code that updates existing blocks to match the IPAM config
- In the code where we do strict affinity checking, just remove the check for `block.StrictAffinity` altogether.

Then, we'll _always_ just be looking at the value in the IPAMConfig struct. WDYT?

Reply:
Makes sense, thanks for the detailed explanation! Will take care of it. Thank you!